src/app/leverage/components/leverage-widget/index.tsx

```diff
@@ -190,16 +114,22 @@ export function LeverageWidget(props: LeverageWidgetProps) {
       />
       <Summary />
       <Fees costOfCarry={costOfCarry} leverageType={leverageType} />
-      <TradeButton
-        label={buttonLabel}
-        isDarkMode
-        isDisabled={isDisabled}
-        isLoading={isFetchingQuote}
-        onClick={onClickButton}
+      <SmartTradeButton
+        contract={contract ?? ''}
+        hasFetchingError={false}
+        hasInsufficientFunds={hasInsufficientFunds}
+        inputTokenAmount={inputTokenAmount}
+        inputToken={inputToken}
+        inputValue={inputValue}
+        isFetchingQuote={isFetchingQuote}
+        isSupportedNetwork={isSupportedNetwork}
+        outputToken={outputToken}
+        buttonLabelOverrides={{
+          [TradeButtonState.default]: 'Review Transaction',
+        }}
+        onOpenTransactionReview={onOpenTransactionReview}
+        onRefetchQuote={() => {}}
       />
-      {buttonState === TradeButtonState.signTerms && (
-        <Warnings warnings={[WarningType.signTerms]} />
-      )}
```

Comment:
took a look at the preview - do we want the other warnings to display on the leverage widget (e.g. MEV protection)? If not, we could use a `supportedWarnings` or similar prop to denote what warnings should get displayed